plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,449 @@
+
+    for extracted_string in extracted_strings:
+      tokens = re.split(self._DELIMITERS_REGEX, extracted_string)
+      output_string = []
+      for token in tokens:
+        if token not in self._DELIMITERS:
+          token = obfuscated_string_map.get(token, token)
+        output_string.append(token)
+      output_strings.append(''.join(output_string))
+
+    return output_strings
+
+  def _DecryptStrings(self, extracted_strings, aes_key):
+    """Decrypt strings extracted from OneDrive logs.
+
+    Encrypted strings are serialized as follows:
+    * padded using PKCS7 then encrypted using AES in CBC mode.  The AES key is
+      from the general.keystore file and a null initialization vector.
+    * base64 encoded where the resulting '/' and '+'' characters are replaced
+      with '_' and '-'' respectively.
+
+    Args:
+      extracted_strings (list[str]): the list of strings extracted from OneDrive
+          log files.
+      aes_key (bytes): the AES key.
+
+    Returns:
+      list[str]: the list of decrypted strings.
+    """
+    if not aes_key:
+      return extracted_strings
+    output_strings = []
+
+    for extracted_string in extracted_strings:
+      tokens = re.split(self._DELIMITERS_REGEX, extracted_string)
+
+      output_string = []
+      for token in tokens:
+        if token in self._DELIMITERS:
+          output_string.append(token)
+          continue
+
+        # minimum block size of base64 encoded AES-CBC is 22 bytes
+        if len(token) < 22:
+          output_string.append(token)
+          continue
+
+        if not all(c in self._BASE64_ALPHABET for c in token):
+          output_string.append(token)
+          continue
+
+        rem = len(token) % 4
+        if rem == 1:
+          output_string.append(token)
+          continue
+
+        while len(token) % 4:
+          token += '='
+
+        fixed_token = token.replace('_', '/').replace('-', '+')
+        ciphertext = base64.b64decode(fixed_token)
+
+        if len(ciphertext) % 16:
+          output_string.append(token)
+          continue
+
+        aes_context = pycaes.context()
+        aes_context.set_key(pycaes.crypt_modes.DECRYPT, aes_key)
+        plaintext = pycaes.crypt_cbc(
+            aes_context, pycaes.crypt_modes.DECRYPT,
+            self._AES_IV, ciphertext)
+
+        unpadder = padding.PKCS7(128).unpadder()
+        try:
+          data = unpadder.update(plaintext) + unpadder.finalize()
+        except ValueError:
+          output_string.append(token)
+          continue
+
+        output_string.append(data.decode('utf-16-le'))
+      output_strings.append(''.join(output_string))
+    return output_strings
+
+  def _ExtractStrings(self, buffer):
+    """Extracts strings from a buffer of bytes.
+
+    Args:
+      buffer (bytearray): the buffer.
+      aes_key (bytes): the AES decryption key from general.keystore.
+      obfuscated_string_map (dict[str, str]): the mapping of obfuscated string
+          values to their true values.
```

Comment:
`true values` => original or unobfuscated values ?

Reply:
Changed to original